src/main/java/com/uber/cadence/internal/replay/DecisionsHelper.java

```diff
@@ -252,6 +267,10 @@ void handleRequestCancelExternalWorkflowExecutionFailed(HistoryEvent event) {
   }
 
   long signalExternalWorkflowExecution(SignalExternalWorkflowExecutionDecisionAttributes signal) {
+    addAllMissingVersionMarker(false, Optional.empty());
+
+    signal.setControl(getNextId().getBytes(StandardCharsets.UTF_8));
```

Comment:
Why do we need it back here?

Reply:
This needs to be done after addAllMissingVersionMarker. It used to be outside of this method and it's moved here..